spring-integration-core/src/main/java/org/springframework/integration/mapping/AbstractHeaderMapper.java

```diff
@@ -423,6 +454,55 @@ public boolean matchHeader(String headerName) {
 			return false;
 		}
 
+		@Override
+		public boolean isNegated() {
+			return false;
+		}
+
+	}
+
+	/**
+	 * A pattern-based {@link HeaderMatcher} that matches if the specified
+	 * header matches the specified simple pattern.
+	 * @see org.springframework.util.PatternMatchUtils#simpleMatch(String, String)
+	 * @since 4.3
+	 */
+	protected static class SinglePatternBasedHeaderMatcher implements HeaderMatcher {
+
+		private static final Log logger = LogFactory.getLog(HeaderMatcher.class);
```

Comment:
Not sure what you mean - they all reference the same object - there is only one object in memory.

Of course, it could be pulled up to the interface so there's only one reference too.